includes/Classifai/Features/Classification.php

```diff
@@ -112,46 +852,158 @@ public function get_feature_default_settings(): array {
+			return $feature_taxonomies;
 		}
 
-		return apply_filters(
-			'classifai_' . static::ID . '_run',
-			$result,
-			$provider_instance,
-			$args,
-			$this
-		);
+		foreach ( array_keys( $provider_instance->nlu_features ) as $feature_name ) {
+			if ( ! get_classification_feature_enabled( $feature_name ) ) {
+				continue;
+			}
+
+			$taxonomy   = get_classification_feature_taxonomy( $feature_name );
+			$permission = check_term_permissions( $taxonomy );
+
+			if ( is_wp_error( $permission ) ) {
+				continue;
+			}
+
+			if ( 'post_tag' === $taxonomy ) {
+				$taxonomy = 'tags';
+			}
+
+			if ( 'category' === $taxonomy ) {
+				$taxonomy = 'categories';
+			}
+
+			$feature_taxonomies[] = $taxonomy;
+		}
+
+		return $feature_taxonomies;
+	}
+
+	/**
+	 * Render the NLU features settings.
+	 *
+	 * @param array $args Settings for the inputs
+	 */
+	public function render_nlu_feature_settings( array $args ) {
+		$feature = $args['feature'];
+		$labels  = $args['labels'];
+
+		$taxonomies = $this->get_supported_taxonomies();
+		$features   = $this->get_settings();
+		$taxonomy   = isset( $features[ "{$feature}_taxonomy" ] ) ? $features[ "{$feature}_taxonomy" ] : $labels['taxonomy_default'];
+
+		// Enable classification type
+		$feature_args = [
+			'label_for'  => $feature,
+			'input_type' => 'checkbox',
+		];
+
+		$threshold_args = [
+			'label_for'     => "{$feature}_threshold",
+			'input_type'    => 'number',
+			'default_value' => $labels['threshold_default'],
+		];
+		?>
+
+		<legend class="screen-reader-text">
+			<?php esc_html_e( 'Classification Taxonomy Settings', 'classifai' ); ?>
+		</legend>
+
+		<p>
+			<?php $this->render_input( $feature_args ); ?>
+			<label for="classifai-settings-<?php echo esc_attr( $feature ); ?>">
+				<?php esc_html_e( 'Enable', 'classifai' ); ?>
+			</label>
+		</p>
+
+		<p>
+			<label for="classifai-settings-<?php echo esc_attr( "{$feature}_threshold" ); ?>">
+				<?php echo esc_html( $labels['threshold'] ); ?>
+			</label><br/>
+			<?php $this->render_input( $threshold_args ); ?>
+		</p>
+
+		<?php if ( NLU::ID === $features['provider'] ) : ?>
+		<p>
+			<label for="classifai-settings-<?php echo esc_attr( "{$feature}_taxonomy" ); ?>">
+				<?php echo esc_html( $labels['taxonomy'] ); ?>
+			</label><br/>
+			<select id="classifai-settings-<?php echo esc_attr( "{$feature}_taxonomy" ); ?>" name="<?php echo esc_attr( $this->get_option_name() ); ?>[<?php echo esc_attr( "{$feature}_taxonomy" ); ?>]">
+				<?php foreach ( $taxonomies as $name => $singular_name ) : ?>
+					<option value="<?php echo esc_attr( $name ); ?>" <?php selected( $taxonomy, esc_attr( $name ) ); ?>>
+						<?php echo esc_html( $singular_name ); ?>
+					</option>
+				<?php endforeach; ?>
+			</select>
+		</p>
+		<?php endif; ?>
+		<?php
+	}
+
+	/**
+	 * Return the list of supported taxonomies
+	 *
+	 * @return array
+	 */
+	public function get_supported_taxonomies(): array {
+		$taxonomies = get_taxonomies( [], 'objects' );
```

Comment:
Currently, when we select embedding as a provider, it shows. Image Tag and Format taxonomy as well. I think we should filter it.

```suggestion
		$taxonomies = get_taxonomies( [ 'show_ui' => true ], 'objects' );
```

Reply:
This is fixed in [b0b3146](https://github.com/10up/classifai/pull/709/commits/b0b3146045bd5bfc6eca728f7f0d221aa1c64e6b). I not only added the `show_ui` argument but also get the value of the post types that are enabled and ensure the taxonomies we show are registered to those post types. This ensures the Image Tag taxonomy, for instance, doesn't show up, as that is only registered to attachments

Reply:
So I actually ended up changing this because of how we register our custom Watson taxonomies. These are only registered if they are chosen as a setting and so `show_ui` is false until that happens. This means if we set `show_ui` to true, those taxonomies will never show up.

Also found a bug that if we pass in the supported post types to our `get_taxonomies` call, it requires all post types be set for that taxonomy. So if someone were to select both Post and Page, categories and tags no longer show up due to not being supported by Pages. I've modified this a bit to where we check post type support after the query, while we loop through our taxonomies

Reply:
This looks correct now. However, Post Format taxonomy was still showing in the settings So, I have explicitly excluded it in https://github.com/10up/classifai/pull/709/commits/420d2f051554b0d3b57b8163fa41d3c1c58579eb

Thank you.